downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/BiologicalProcessAnnotationBuilder.java

```diff
@@ -124,11 +128,13 @@ private static void processProteins(Set<GKInstance> proteinInstances, GKInstance
      * @param taxonIdentifier -- String, CrossReference ID of protein's species. Example: 9606 (Human Species crossReference identifier from NCBI)
      * @throws Exception -- MySQLAdaptor exception.
      */
-    private static void getGOBiologicalProcessLine(GKInstance referenceEntityInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
+    private static String getGOBiologicalProcessLine(GKInstance referenceEntityInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
+        String goaLine = null;
         for (Map<String, String> biologicalProcessAccession : getGOBiologicalProcessAccessions(reactionInst, 0)) {
-            String goaLine = GOAGeneratorUtilities.generateGOALine(referenceEntityInst, BIOLOGICAL_PROCESS_LETTER, biologicalProcessAccession.get(ACCESSION_STRING), biologicalProcessAccession.get(EVENT_STRING), TRACEABLE_AUTHOR_STATEMENT_CODE, taxonIdentifier);
+            goaLine = GOAGeneratorUtilities.generateGOALine(referenceEntityInst, BIOLOGICAL_PROCESS_LETTER, biologicalProcessAccession.get(ACCESSION_STRING), biologicalProcessAccession.get(EVENT_STRING), TRACEABLE_AUTHOR_STATEMENT_CODE, taxonIdentifier);
```

Comment:
Since this is re-assigned on each iteration of the loop, wouldn't it be the only the last iteration's value would be returned at the end of the function?

Reply:
Good catch, I've changed the method to have a List returned